exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillConvertletTable.java

```diff
@@ -138,14 +105,209 @@ public SqlRexConvertlet get(SqlCall call) {
       ((SqlBasicCall) call).setOperator(wrapper);
       return sqlRexConvertlet;
     }
-
-    if ((convertlet = map.get(call.getOperator())) != null) {
+    if ((convertlet = operatorToConvertletMap.get(call.getOperator())) != null) {
       return convertlet;
     }
-
     return StandardConvertletTable.INSTANCE.get(call);
   }
 
-  private DrillConvertletTable() {
+  /**
+   * Custom convertlet to handle extract functions. Optiq rewrites
+   * extract functions as divide and modulo functions, based on the
+   * data type. We cannot do that in Drill since we don't know the data type
+   * till we start scanning. So we don't rewrite extract and treat it as
+   * a regular function.
+   */
+  private SqlRexConvertlet extract() {
```

Comment:
I agree that static missed here but I would like to leave naming as is. The name is short and the return type is clear.

Reply:
Ok, let's leave method names as they are. My objection regarding their names appeared because when I was reviewing this PR, it wasn't unclear what these methods do (or return) in the places where they are used, so I advised renaming them.

Reply:
ok, renamed. Please let me know whether it is clear now.